tests/ord-service/tests/api_test.go

```diff
@@ -166,321 +120,374 @@ func TestORDService(t *testing.T) {
-			require.True(t, exists)
-			require.False(t, events[i].Get("description").Exists())
-		}
-	})
+				require.Equal(t, *expectedAPI.Description, apis[i].Get("description").String())
+			}
+		})
 
-	//Ordering:
-	t.Run("Requesting ordering of Bundles returns them as expected", func(t *testing.T) {
-		escapedOrderByValue := urlpkg.PathEscape("title asc,description desc")
-		respBody := makeRequest(t, httpClient, fmt.Sprintf("%s/bundles?$orderby=%s&$format=json", testConfig.ORDServiceURL, escapedOrderByValue))
-		require.Equal(t, 1, len(gjson.Get(respBody, "value").Array()))
-		require.Equal(t, appInput.Bundles[0].Name, gjson.Get(respBody, "value.0.title").String())
-		require.Equal(t, *appInput.Bundles[0].Description, gjson.Get(respBody, "value.0.description").String())
-	})
+		t.Run(fmt.Sprintf("Requesting ordering of Bundle Events for tenant %s returns them as expected", testData.tenant), func(t *testing.T) {
+			escapedOrderByValue := urlpkg.PathEscape("title asc,description desc")
+			respBody := makeRequestWithHeaders(t, httpClient, fmt.Sprintf("%s/bundles?$expand=events($orderby=%s)&$format=json", testConfig.ORDServiceURL, escapedOrderByValue), map[string][]string{tenantHeader: {testData.tenant}})
 
-	t.Run("Requesting ordering of Bundle APIs returns them as expected", func(t *testing.T) {
-		escapedOrderByValue := urlpkg.PathEscape("title asc,description desc")
-		respBody := makeRequest(t, httpClient, fmt.Sprintf("%s/bundles?$expand=apis($orderby=%s)&$format=json", testConfig.ORDServiceURL, escapedOrderByValue))
+			events := gjson.Get(respBody, "value.0.events").Array()
+			require.Len(t, events, len(testData.appInput.Bundles[0].EventDefinitions))
 
-		apis := gjson.Get(respBody, "value.0.apis").Array()
-		require.Len(t, apis, len(appInput.Bundles[0].APIDefinitions))
+			for i := range testData.appInput.Bundles[0].EventDefinitions {
+				name := events[i].Get("title").String()
+				require.NotEmpty(t, name)
 
-		for i := range appInput.Bundles[0].APIDefinitions {
-			name := apis[i].Get("title").String()
-			require.NotEmpty(t, name)
+				expectedEvent, exists := testData.eventsMap[name]
+				require.True(t, exists)
 
-			expectedAPI, exists := apisMap[name]
-			require.True(t, exists)
+				require.Equal(t, *expectedEvent.Description, events[i].Get("description").String())
+			}
+		})
+	}
 
-			require.Equal(t, *expectedAPI.Description, apis[i].Get("description").String())
-		}
+	t.Run("404 when request to ORD Service for api spec have another tenant header value", func(t *testing.T) {
+		makeRequestWithHeadersAndStatusExpect(t, httpClient, fmt.Sprintf(testConfig.ORDServiceURL+"/api/%s/specification", apiDefinitionIDDefaultTenant), map[string][]string{tenantHeader: {testConfig.Tenant}}, http.StatusNotFound)
+		makeRequestWithHeadersAndStatusExpect(t, httpClient, fmt.Sprintf(testConfig.ORDServiceURL+"/api/%s/specification", apiDefinitionIDDefaultTenant), map[string][]string{tenantHeader: {testConfig.DefaultTenant}}, http.StatusOK)
```

Comment:
It would be good to add a test for each OData entity (API, Eevent, Bundle) that usage of wrong tenant returns empty array.